File: src/schemas/capabilities.py
# -*- coding: utf-8 -*-
"""Runtime capability response schemas."""

from __future__ import annotations

from pydantic import BaseModel


class AShareIntelligenceCapability(BaseModel):
    enabled: bool = False
    provider_installed: bool = False
    report_enabled: bool = False
    agent_tools_enabled: bool = False
    scoring_enabled: bool = False


class RuntimeCapabilities(BaseModel):
    ashare_intelligence: AShareIntelligenceCapability

File: apps/dsa-web/src/api/capabilities.ts
import apiClient from './index';
import { toCamelCase } from './utils';

export type AShareIntelligenceCapability = {
  enabled: boolean;
  providerInstalled: boolean;
  reportEnabled: boolean;
  agentToolsEnabled: boolean;
  scoringEnabled: boolean;
};

export type RuntimeCapabilities = {
  ashareIntelligence: AShareIntelligenceCapability;
};

export const capabilitiesApi = {
  async getCapabilities(): Promise<RuntimeCapabilities> {
    const response = await apiClient.get<Record<string, unknown>>('/api/v1/capabilities');
    return toCamelCase<RuntimeCapabilities>(response.data);
  },
};
